client/my-sites/upgrades/domain-management/transfer/icann-verification.jsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import SectionHeader from 'components/section-header';
+import { resendIcannVerification } from 'lib/upgrades/actions/domain-management';
+import Button from 'components/button';
+import notices from 'notices';
+
+const IcannVerification = React.createClass( {
+	getInitialState() {
+		return {
+			submitting: false
+		};
+	},
+
+	handleClick() {
+		this.setState( { submitting: true } );
+
+		resendIcannVerification( this.props.selectedDomainName, ( error ) => {
+			if ( error ) {
+				notices.error( error.message );
+			} else {
+				notices.success( this.translate(
+					'We successfully resent the ICANN verification email to your ' +
+					'email address on file.'
+				) );
+			}
+
+			this.setState( { submitting: false } );
+		} );
+	},
+
+	render() {
+		return (
+			<div>
+				<SectionHeader label={ this.translate( 'Transfer Domain' ) }>
+					<Button
+						onClick={ this.handleClick }
+						disabled={ this.state.submitting }
+						compact
+						primary={ true }>{ this.translate( 'Resend Verification Email' ) }</Button>
```

Comment:
I didn't want to nitpick this, but since we have `compact` just above taking advantage of how JSX handles [boolean expressions](https://facebook.github.io/react/docs/jsx-in-depth.html#boolean-attributes), it might be good to stick to one convention, for consistency sake. So either explicit `true` value or an implicit one :)